### ISSUE TITLE
Allow any Python version if using Docker image for Lambda

### ISSUE THREAD

**Issue by gaborschulz** [zappa/Zappa#1140]
## Context
AWS allows users to work with custom images. These custom images can use any version of Python, not just the predefined ones, like 3.7, 3.8 or 3.9. E.g. I could build my own image using Python 3.10 or 3.11alpha and use that custom image for deployment. However, even though this is a completely valid setup, Zappa does not allow this to happen if I have a virtual environment with Python 3.10 or 3.11.

## Expected Behavior
Allow the usage of any Python version if I am using a custom image for deployment.

## Actual Behavior
The deployment fails because of an unsupported Python version.

## Possible Fix
Skip the environment version check if a custom image is used for deployment

## Steps to Reproduce
1. Create a Python 3.10 environment
2. Use a custom Docker image for deployment with Python 3.10
3. While this is a completely valid setup, Zappa deployment will fail

## Your Environment
* Zappa version used: 0.54.1
* Operating System and Python version: Linux, Python 3.10

**Comment by monkut** [comment]
Zappa supports 3.7 through 3.9.  So you'll need to run zappa from one of those environments, however, I can't seem to find a restriction anywhere on the version used in docker.

Can you share the error?

**Comment by gaborschulz** [comment]
That seems to be the actual issue. I would need to have an environment on Python 3.10 for development and building the container. Then, I would need to switch the environment to a 3.9 just to run `zappa update` and then switch back to the 3.10 environment to continue developing. That becomes very cumbersome, eventually. 

In understand that it would be a relatively complex refactoring as the version check is actually done in the `__init__.py` and performed before anything else is done. 

It would be great to move this to another section of the code and only perform it if the configuration does not specify a Docker image.

**Comment by monkut** [comment]
@gaborschulz I've attempted to add a solution for this case in the PR below.

Let me know if you believe this will resolve your issue.

https://github.com/zappa/Zappa/pull/1180

**Comment by gaborschulz** [comment]
@monkut, that's amazing, I think that's exactly the fix I was looking for. Thank you so much.

There's only one minor thing that would need to be checked:

Pip won't let Zappa be installed inside the Docker container if it's higher than Python 3.9 because of this line in `setup.py`:
```
python_requires=">=3.7, <3.10",
```

Is it possible to change this line to 
```
python_requires=">=3.7",
```

**Comment by monkut** [comment]
@gaborschulz Thanks for the feedback, updated!

**Comment by monkut** [comment]
@gaborschulz Released in version 0.56.0.  If you have time give it a try and see if it's working for you.

**Comment by gaborschulz** [comment]
@monkut, I've tested the solution, and it's great in most cases, but can behave unpredictable under some circumstances that I was not able to pinpoint. This leads to some build pipelines failing randomly.

I propose an alternative approach. I would assume that folks who build their own Lambda runtime images are advanced users and would be willing to set an environment variable if they wanted to bypass the version check.

My proposal is to set an env var called `ZAPPA_RUNNING_IN_DOCKER` to a truth value. This can be done in the Dockerfile and also in the running environment. I've created a new pr:

[#1204](https://github.com/zappa/Zappa/pull/1204)
